samples/geolocation-weather/src/index.ts

```diff
@@ -0,0 +1,98 @@
+// For AutoRouter documentation refer to https://itty.dev/itty-router/routers/autorouter
+import { AutoRouter } from 'itty-router';
+import { getClientAddressFromRequest, cleanupIpAddress } from "./helpers";
+import { Variables } from "@fermyon/spin-sdk";
+
+
+let router = AutoRouter();
+
+// Route ordering matters, the first route that matches will be used
+// Any route that does not return will be treated as a middleware
+// Any unmatched route will return a 404
+router
+    .get("/", getWeather);
+
+//@ts-ignore
+addEventListener('fetch', async (event: FetchEvent) => {
+    event.respondWith(router.fetch(event.request));
+});
```

Comment:
I think we should check for the variable here and if it does not exist, we should return early 
```suggestion
addEventListener('fetch', async (event: FetchEvent) => {
   let token = Variables.get("waqi_api_token");
   if (!token) {
     event.respondWith(new Response("Internal Server Error", {status: 500});
   }
    event.respondWith(router.fetch(event.request, {token}));
});
```

And in router.get you can access it like 

```
router.get("/", (request, {token}) => getWeather(request, token)) 
```

Reply:
updated with your feedback. Thanks @karthik2804 !